addons/transitions/Transitions.gd

```diff
@@ -150,16 +148,21 @@ func _common_post_fade(data:Array, new_scene) -> void:
 	if fade_scene in _root:
 		_root.remove_child(fade_scene)
 
-func _set_scene(new_scene):
+func _set_scene(new_scene:Node):
 	# Dispose old scene so we don't get any camera jitters or wierdness.
-	var previous_scene = _get_current_scene()
 	
-	scene_container.remove_child(previous_scene)
-	print("killed previous scene %s" % previous_scene)
-	previous_scene.queue_free()
+	var parent_node = get_tree().current_scene
```

Comment:
I'm not really clear on what these changes are doing. Can you explain?

Also, `get_tree().current_scene` doesn't always work. In my current project, the resulting scene is a `Tween`, which causes a big in transitioning; it's better to use `_get_current_scene` instead. Can you explain why you decided not to use it, and instead reference `get_tree().current_scene()`?